common/ferrostar/src/algorithms.rs

```diff
@@ -27,6 +26,36 @@ use std::time::SystemTime;
 #[cfg(all(test, feature = "web-time"))]
 use web_time::SystemTime;
 
+/// Get the index of the closest point in the line.
```

Comment:
This docstring needs a bit more detail ;) In particular, it's not clear what `skip_to_index` does and why you'd use it.

It must also document edge cases pending resolution.